### PR TITLE
Feat/form v2

### PR DESCRIPTION
**Pull request by BillyG83**
-  feat(consts): added reg exp helpers
-  feat(Form): validation, style fixes, stories, tests
-  feat(TextArea): validation, style fixes, stories, tests
-   feat(Radios): validation, style fixes, stories, tests
-  feat(Checkbox): validation, style fixes, updated tests and stories
-  feat(Checkbox): validation, style fixes, SVG removed
-  fix(Avatar): isRounded prop optional

### CODE REVIEW

File: src/components/Checkbox/Checkbox.test.tsx [hunk 1]
```diff
@@ -475,28 +497,31 @@ describe('Checkbox - Switch disabled', () => {
...
-        label && fireEvent.click(label);
-        expect(console.log).not.toHaveBeenCalled();
-    });
+    // it('the component is disabled so will not return anything', () => {
+    //     const label: HTMLLabelElement | null = container.querySelector(
+    //         `[data-testid="${labelTestID}"]`,
+    //     );
+    //     label && fireEvent.click(label);
+    //     expect(testEvent).not.toHaveBeenCalled();
+    // });
```
Comment (locothedev) [inline src/components/Checkbox/Checkbox.test.tsx:506]: commmeeeeeeentssss

File: src/components/Form/Form.styles.ts [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
+import resetCSS from '../../styles/reset';
 
 export const TitleStyled = styled.h3`
-	${resetCSS}
-	${fonts.heading}
+    ${resetCSS}
+    ${fonts.heading}
 	${fonts.h3}
 	margin-bottom: 8px;
 `;
```
Comment (locothedev) [inline src/components/Form/Form.styles.ts:10]: prettieeeeer

File: src/components/Form/Form.tsx [hunk 3]
```diff
@@ -7,27 +7,29 @@ import { Radios } from '../Radios';
...
-const Form: React.FC<FormProps> = ({
-    data,
-    id = String(Date.now()),
-    title,
-}) => {
+const Form: React.FC<FormProps> = ({ data, id, title, onSubmit }) => {
     const formSubmitted = (event: React.SyntheticEvent) => {
         event.preventDefault();
         event.stopPropagation();
-        console.log(data[5].selected);
```
Comment (locothedev) [inline src/components/Form/Form.tsx:23]: Is console.log really needed here?

File: src/components/Checkbox/Checkbox.styles.ts [hunk 4]
```diff
@@ -138,3 +137,18 @@ export const switchOnStyles = css`
...
+
+export const StyledLabel = styled.label<
+    Pick<CheckboxProps, 'layout' | 'checked' | 'disabled'>
+>`
+    ${labelStyles}
+    ${(p) => p.disabled && labelDisabled}
+	${(p) => p.layout === 'box' && boxStyles}
+    ${(p) => p.layout === 'box' && p.checked && boxCheckedStyles}
+	${(p) => p.layout === 'switch' && switchStyles}
+    ${(p) => p.layout === 'switch' && p.checked && switchOnStyles}
```
Comment (locothedev) [inline src/components/Checkbox/Checkbox.styles.ts:153]: prettttieeerrr

File: src/components/TextArea/types.tsx [hunk 5]
```diff
@@ -44,4 +46,9 @@ export interface TextAreaProps {
      */
     value?: string;
 
+    /**
+     * You can validate your textarea
+     * required, characterMinLength, characterMaxLength, numberMin, numberMax, regExp , regExpInvalidMessage
+     */
+    validation?: ValidateInput;
```
Comment (locothedev) [inline src/components/TextArea/types.tsx:53]: Alphabetically strikes again . This time very harsh lol

File: src/components/Input/Input.tsx [hunk 6]
```diff
@@ -150,7 +143,7 @@ const Input: React.FC<InputProps> = ({
                     hasPrefix={typeof prefixIcon !== 'undefined'}
```
Comment (locothedev) [inline src/components/Input/Input.tsx:143]: isnt prefixIcon enough in this case?

File: src/components/Form/Form.styles.ts (hunk 2 above)
Comment (BillyG83) [reply src/components/Form/Form.styles.ts:10]: it let me down again, its not flawless for sure 😢

**Pull request merged**